packages/react-internal/src/components/ResizeGroup/ResizeGroup.base.tsx

```diff
@@ -309,63 +309,61 @@ const hiddenDivStyles: React.CSSProperties = { position: 'fixed', visibility: 'h
 const hiddenParentStyles: React.CSSProperties = { position: 'relative' };
 const COMPONENT_NAME = 'ResizeGroup';
 
+type ResizeDataAction = {
+  type: 'resizeData' | keyof IResizeGroupState;
+  value: IResizeGroupState[keyof IResizeGroupState] | IResizeGroupState;
+};
+
+function resizeDataReducer(state: IResizeGroupState, action: ResizeDataAction): IResizeGroupState {
```

Comment:
Can you add comments inline about the reason for this optimization (basically what you wrote in the PR description)?

Reply:
Done!